### PR TITLE
Check that Expr is none before adding fixup

### LINKED ISSUE #17463: False Positive on Missing Comma
**rust-analyzer version**: 
rust-analyzer version: 0.0.0 (e08f7953f 2024-06-20) [/media/Other/ProgrammingProjects/RustProjects/rust-analyzer/target/release/rust-analyzer]
I built this based on the latest commit today. 

**rustc version**: (eg. output of `rustc -V`)
rustc 1.79.0 (129f3b996 2024-06-10)

**editor or extension**: VSCode 1.90.1 OpenSuse Tumbleweed Linux 6.9.5

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)
[VS Code Settings](https://gist.github.com/wyatt-herkamp/23447c21b0550750be06ae080517cc0f)
**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))
Private Repository however I was able to make a simple reproduction. 
**code snippet to reproduce**: 
```rust
use actix_web::{
    post,
    web::{self, Data},
    HttpResponse,
};
use sea_orm::{prelude::*, ActiveValue, DatabaseConnection};

fn main() {
    println!("Hello, world!");
}

use serde::{Deserialize, Serialize};
#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "users")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,
    pub name: String,
    pub username: String,
    pub email: String,
    #[serde(skip_serializing)]
    pub password: String,
    pub admin: bool,
    #[sea_orm(default_expr = "Expr::current_timestamp()")]
    pub created: DateTimeWithTimeZone,
}
impl ActiveModelBehavior for ActiveModel {}
#[derive(Deserialize)]
pub struct NewUser {
    pub username: String,
    pub password: String,
    pub email: String,
    pub name: String,
}
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

#[post("/install")]
pub async fn install(
    post: web::Json<NewUser>,
    database: Data<DatabaseConnection>,
) -> Result<HttpResponse, actix_web::error::Error> {
    let post = post.into_inner();
    let user = ActiveModel {
        username: ActiveValue::Set(post.username),
        password: ActiveValue::Set(post.password),
        email: ActiveValue::Set(post.email),
        name: ActiveValue::Set(post.name),
        admin: ActiveValue::Set(true),
        ..Default::default()
    };
    Entity::insert(user).exec(database.as_ref()).await.unwrap();

    Ok(HttpResponse::Ok().finish())
}
```
image

### So when I expand the post macro I get this. 
image

I was going to try to fix it however, I have no idea where it is coming from. If someone has an idea where it is coming from let me know I will try to fix it myself. 

If I move to the stable version of rust-analyzer so the release on 17th I have no issues. But the nightly release has issues.

### PR DESCRIPTION
**Pull request by wyatt-herkamp**
Closes #17463

### CODE REVIEW

**Comment by lnicola** [comment]
Needs a `cargo fmt`.

**Comment by wyatt-herkamp** [comment]
> Needs a `cargo fmt`.

Does not change anything? 



> And can you also include the test?

The test @Veykril commented fails. It says the `__ra_fixup` is not the expected result.

File: crates/hir-expand/src/fixup.rs [hunk 1]
```diff
@@ -277,7 +277,7 @@ pub(crate) fn fixup_syntax(
                 },
                 ast::RecordExprField(it) => {
                     if let Some(colon) = it.colon_token() {
-                        if it.name_ref().is_some() {
+                        if it.name_ref().is_some() && it.expr().is_none(){
```
Comment (lnicola) [inline crates/hir-expand/src/fixup.rs:280]: Right, it doesn't see through `match_ast!`.

```suggestion
                        if it.name_ref().is_some() && it.expr().is_none() {
```

Comment (wyatt-herkamp) [reply crates/hir-expand/src/fixup.rs]: Yeah I corrected it

**Comment by lnicola** [comment]
Yeah, I meant adding a:

```rust
    #[test]
    fn no_fixup_record_ctor_field() {
        check(
            r#"
fn foo() {
    R { f: a }
}
"#,
            expect![[r#"
fn foo () {R {f : a}}
"#]],
        )
    }
```

(untested)

But I'm not sure it's worth the hassle.

**Comment by wyatt-herkamp** [comment]
> Yeah, I meant adding a:
> 
> ```rust
>     #[test]
>     fn no_fixup_record_ctor_field() {
>         check(
>             r#"
> fn foo() {
>     R { f: a }
> }
> "#,
>             expect![[r#"
> fn foo () {R {f : a}}
> "#]],
>         )
>     }
> ```
> 
> (untested)
> 
> But I'm not sure it's worth the hassle.

I added it. It doesn't hurt and it passes.

File: crates/hir-expand/src/fixup.rs [hunk 2]
```diff
@@ -839,6 +839,19 @@ fn foo() {
 "#,
             expect![[r#"
 fn foo () {R {f : __ra_fixup}}
+"#]],
+        )
+    }
+    #[test]
```
Comment (lnicola) [inline crates/hir-expand/src/fixup.rs:845]:
```suggestion

    #[test]
```

**Comment by wyatt-herkamp** [comment]
I squashed all of them because git signing broke for the first two commits.

**Pull request merged**